python/kvikio/_lib/libkvikio.pyx

```diff
@@ -12,7 +12,28 @@ from libcpp.utility cimport move, pair
 
 from . cimport kvikio_cxx_api
 from .arr cimport Array
-from .kvikio_cxx_api cimport FileHandle, future, is_future_done
+from .kvikio_cxx_api cimport (
+    FileHandle,
+    StreamFuture,
+    cudaStream_t,
+    future,
+    is_future_done,
+)
+
+
+cdef class PyStreamFuture:
```

Comment:
I think we should rename this to something like `IOFutureStream`. Or maybe, we can extend `IOFuture` to also have an optional stream?

Reply:
Good points and I've done the rename! The extended class seems a bit overloaded. Maybe we can think about how to combine the two classes in later PRs? I am slacking here :D